### ISSUE TITLE
ParticleStateDistribution?

### ISSUE THREAD

**Issue by ebalaban** [JuliaPOMDP/POMDPToolbox.jl#6]
Can we add the following type to POMDPToolbox.jl?
```julia
type ParticleStateDistribution{T} <: AbstractDistribution
    particles::Vector{Particle{T}}
end
```
It would be similar to `ParticleBelief` (that's already in there) and would allow me to keep things cleaner, given the latest API changes. Having this type in a common package, I can use it in the problem's `initial_state_distribution()` and also in the updater, to define `initialize_belief()`, without needing to implement ugly hooks from the problem into the updater or vice versa.

**Comment by zsunberg** [comment]
Yeah, we should definitely have something like this! Neither @etotheipluspi or I know where `ParticleBelief` actually came from, and it shouldn't be in `discrete.jl` so please remove that when you get a chance.

If you want, we can still call it `ParticleBelief` - a belief can certainly be an `AbstractDistribution`. If you don't want to call it a belief, I would vote for shortening it to `ParticleDistribution`. Also, I would vote for making it a subtype of the parameterized `AbstractDistribution{T}` rather than just `AbstractDistribution`

**Comment by mykelk** [comment]
+1 on everything @zsunberg said.

**Comment by ebalaban** [comment]
Sounds great to me. How about the following:
```julia
type Particle{T}
    state::T
    weight::Float64
end

type ParticleBelief{T} <: AbstractDistribution{T}
    particles::Vector{Particle{T}}
end

typealias ParticleDistribution{T} ParticleBelief{T}
```

I put ParticleBelief in there originally (issue #2), to go along with Particle. At the time `discrete.jl` looked like the most appropriate option (or, rather, the least inappropriate option). Should we put all of this in a new file, `particles.jl`?

**Comment by zsunberg** [comment]
Yeah, or maybe just `particle.jl` because particle is an adjective, i.e.
the file contains particle beliefs.

**Comment by ebalaban** [comment]
Sure, `particle.jl` is fine with me.

**Comment by ebalaban** [comment]
Implemented with  c291932.
